improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -395,7 +415,60 @@ def test_old_naming_convention(self):
                         t_min=(1. - WeatherSymbols().float_tolerance),
                         t_max=(1. + WeatherSymbols().float_tolerance)))
         self.assertIsInstance(result, list)
-        self.assertIsInstance(result[0], str)
+        self.assertTrue(all([isinstance(s, str) for s in result]))
+        self.assertEqual(len(result), 1)
+        self.assertEqual(result[0], expected)
+
+    def test_complex_condition(self):
+        """Test with a condition that uses an operator"""
+        query = {
+            'rain_or_snow': self.dummy_queries['significant_precipitation'],
+        }
+        query['rain_or_snow']['diagnostic_fields'] = [
+            [
+                'probability_of_lwe_sleetfall_rate_above_threshold',
+                '-',
+                'probability_of_rainfall_rate_above_threshold',
+            ],
+            [
+                'probability_of_lwe_sleetfall_rate_above_threshold',
+                '-',
+                'probability_of_lwe_snowfall_rate_above_threshold',
+            ],
+        ]
+        query['rain_or_snow']['diagnostic_thresholds'] = [
+            [AuxCoord(0.1, units='mm hr-1'),
+             AuxCoord(0.1, units='mm hr-1')],
+            [AuxCoord(0.1, units='mm hr-1'),
+             AuxCoord(0.1, units='mm hr-1')]]
+        query['rain_or_snow']['diagnostic_conditions'] = [
+            ['above', 'above'],
+            ['above', 'above'],
+        ]
+        plugin = WeatherSymbols()
+        test_condition = query['rain_or_snow']
+        result = plugin.create_condition_chain(test_condition)
+        print(result)
```

Comment:
Can remove the print statement.

Reply:
Oops. That's one reason for having a review!